File: BinaryTreePreorder.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        ArrayList<Integer> result = new ArrayList<Integer>();
        helper(root, result);
        return result;
    }
    
    private void helper(TreeNode root, ArrayList<Integer> result) {
        if (root == null) {
            return;
        }
        result.add(root.val);
        
        if (root.left != null) {
            helper(root.left, result);
        }
        
        if (root.right != null) {
            helper(root.right, result);
        }
    }
}

File: Python/BinaryTreePreorder.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def preorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        list = [];
        self.helper(root, list);
        return list;
        
    def helper(self, root, result):
        if root is None:
            return;
        result.append(root.val);
        self.helper( root.left, result);
        self.helper( root.right, result);
        

